File: Array/832-Flipping-an-Image/832-Flipping-an-Image.js
/**
 * @param {number[][]} image
 * @return {number[][]}
 */
var flipAndInvertImage = function(image) {
    return image.map(arr => {
        arr = flip(arr);
        for(let i = 0;i<arr.length;i++) {
            arr[i] ^= 1;
        }

        return arr;
    })
};

const flip = (arr) => {
    let start = 0, end = arr.length - 1;
    while(start < end) {
        let temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
        start++;
        end--;
    }
    return arr;
}

File: Array/832-Flipping-an-Image/832-Flipping-an-Image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        for(int[] arr : image) {
            flip(arr);
            for(int i=0;i<arr.length;i++) {
                arr[i] ^= 1;
            }
        }
        return image;
    }
    public void flip(int[] arr) {
        int start = 0, end = arr.length - 1;
        while(start < end) {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
    }
}
